src/common/snippets/src/op/perf_count.cpp

```diff
@@ -3,10 +3,112 @@
 //
 #ifdef SNIPPETS_DEBUG_CAPS
 
+#include <fstream>
+
 #include "snippets/op/perf_count.hpp"
 
 namespace ov {
 namespace snippets {
+
+//////////////////utils///////////////
+
+namespace utils {
+
+//////////////////utils::Dumper///////////////
+
+void Dumper::init(const std::string &params) {
+    m_params = params;
+}
+
+//////////////////utils::ConsoleDumper///////////////
+
+ConsoleDumper::~ConsoleDumper() {
+}
+
+void ConsoleDumper::update(const op::PerfCountEnd* node) {
+    auto accumulation = node->get_accumulation();
+    auto iteration = node->get_iteration();
+    OPENVINO_ASSERT(accumulation.size() == iteration.size(), "accumulation size should be the same as iteration size in perf_count_end node.");
+    auto iterator_iter = iteration.begin();
+    auto iterator_acc = accumulation.begin();
+    int t_num = 0;
+    uint64_t avg_max = 0;
+    std::cout << "Perf count data in perfCountEnd node with name " << node->get_friendly_name() << " is:" << std::endl;
+    for (; iterator_iter != iteration.end(); ++iterator_iter, ++iterator_acc) {
+        const auto iter = *iterator_iter;
+        const auto acc = *iterator_acc;
+        uint64_t avg = iter == 0 ? 0 : acc / iter;
+        if (avg > avg_max)
+            avg_max = avg;
+        std::cout << "accumulated time:" << acc << "ns, iteration:" << iter << " avg time:" << avg << "ns" << " on thread:" << t_num << std::endl;
+        t_num++;
+    }
+
+    // max time of all threads: combine for reduce max
+    auto BinaryFunc = [](const uint64_t& a, const uint64_t& b) {
+        return a >= b ? a : b;
+    };
+
+    // max accumulation
+    uint64_t acc_max = accumulation.combine(BinaryFunc);
+    std::cout << "max accumulated time:" << acc_max << "ns" << std::endl;
+    // max avg
+    std::cout << "max avg time:" << avg_max << "ns" << std::endl;
```

Comment:
I see that you decided to retain the old behavior, where the output is printed on every iteration instead of printing it once in the destructor. Is there any particular reason why?
As we discussed on the meeting, this approach have higher runtime overheads that might be important when we have nested perf counters. Moreover now we have two dumpers with mismatching behavior, which might be confusing.

Reply:
No reason, actually. Moved to destructor, aligned behavior